### PR TITLE
add ability to provide headers on request rejection

### PR DESCRIPTION
**Pull request by artgon**

### CODE REVIEW

File: zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java [hunk 1]
```diff
@@ -93,10 +94,11 @@ public static void rejectByClosingConnection(
      */
     public static void sendRejectionResponse(
             ChannelHandlerContext ctx, StatusCategory nfStatus, String reason, HttpRequest request,
-            @Nullable Integer injectedLatencyMillis, HttpResponseStatus rejectedCode, String rejectedBody) {
+            @Nullable Integer injectedLatencyMillis, HttpResponseStatus rejectedCode, String rejectedBody,
```
Comment (carl-mastrangelo) [inline zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java:97]: need to update the javadoc

File: zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java [hunk 2]
```diff
@@ -132,7 +134,8 @@ public static void allowThenClose(ChannelHandlerContext ctx) {
      */
     public static void handleRejection(
             ChannelHandlerContext ctx, Object msg, RejectionType rejectionType, StatusCategory nfStatus, String reason,
-            @Nullable Integer injectedLatencyMillis, HttpResponseStatus rejectedCode, String rejectedBody)
+            @Nullable Integer injectedLatencyMillis, HttpResponseStatus rejectedCode, String rejectedBody,
```
Comment (carl-mastrangelo) [inline zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java:137]: same comment

File: zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java [hunk 3]
```diff
@@ -220,6 +235,7 @@ private static FullHttpResponse createRejectionResponse(
         if (closeConnection) {
             headers.set(HttpHeaderNames.CONNECTION, "close");
         }
+        rejectionHeaders.forEach(headers::add);
```
Comment (carl-mastrangelo) [inline zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java:238]: headers.addAll(rejectionHeaders.forEach); 

?

Comment (artgon) [reply zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java:238]: I don't see an `addAll` method on that object.

File: zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java [hunk 4]
```diff
@@ -162,6 +178,14 @@ public static void handleRejection(
         }
     }
 
+
+    public static void reject(
```
Comment (carl-mastrangelo) [inline zuul-core/src/main/java/com/netflix/netty/common/throttle/RejectionUtils.java:182]: Later linters nag on missing javadoc for public methods,  consider copying some of the existing docs for the param names.

**Pull request merged**